src/services/IPRService.ts

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-console */
+import { Logger } from "@aws-lambda-powertools/logger";
+import { AppError } from "../utils/AppError";
+import { DynamoDBDocument, GetCommand, UpdateCommand } from "@aws-sdk/lib-dynamodb";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { IpvStartedOnEvent, JourneyWentAsyncOnEvent, ReadyToResumeOnEvent, AccountDeletedOnEvent } from "../models/IPREventTypes";
+
+export class IPRService {
+	readonly tableName: string;
+
+	private readonly dynamo: DynamoDBDocument;
+
+	readonly logger: Logger;
+
+	private static instance: IPRService;
+
+	constructor(tableName: any, logger: Logger, dynamoDbClient: DynamoDBDocument) {
+		this.tableName = tableName;
+		this.dynamo = dynamoDbClient;
+		this.logger = logger;
+	}
+
+	static getInstance(tableName: string, logger: Logger, dynamoDbClient: DynamoDBDocument): IPRService {
+		if (!IPRService.instance) {
+			IPRService.instance = new IPRService(tableName, logger, dynamoDbClient);
+		}
+		return IPRService.instance;
+	}
+
+	async isFlaggedForDeletion(userId: string): Promise<boolean | undefined> {
+		this.logger.debug("Table name " + this.tableName);
+		const getSessionCommand = new GetCommand({
+			TableName: this.tableName,
+			Key: {
+				userId,
+			},
+		});
+
+		let session;
```

Comment:
`session` only seems to be used inside the try block, could we move the declaration inside the try/catch & mark it was constant?

Reply:
Will update initially the check if session exists was outside the try catch block hence why it's like this, I should've tidied it up